me.id.webverify/app/src/main/java/me/id/meidwebverify/MainActivity.java

```diff
@@ -35,42 +43,72 @@ public void onClick(View v) {
    * Method that Starts the Verification Process.
    */
   public void verify() {
-    Spinner spnRoute = (Spinner) findViewById(R.id.spnRoute);
     Spinner propRoute = (Spinner) findViewById(R.id.spnProperties);
-    String affiliationType = "";
 
     returnProperties = propRoute.getSelectedItem().toString().equals("Yes");
 
+    final IDmeScope affiliationType = getSelectedAffiliationType();
+    if (affiliationType == null) {
+      showError(new IllegalStateException("Affiliation Type is required"));
+      return;
+    }
+    IDmeWebVerify.getInstance().getAccessToken(affiliationType, new IDmeGetAccessTokenListener() {
+      @Override
+      public void onSuccess(String accessToken) {
+        if (returnProperties) {
+          showUserProfileInformation(affiliationType);
+        } else {
+          showResponse(accessToken);
+        }
+      }
+
+      @Override
+      public void onError(Throwable throwable) {
+        showError(throwable);
+      }
+    });
+  }
+
+  @Nullable
+  private IDmeScope getSelectedAffiliationType() {
+    Spinner spnRoute = (Spinner) findViewById(R.id.spnRoute);
     Object selectedItem = spnRoute.getSelectedItem();
     if (selectedItem == null) {
-      Toast.makeText(this, "Affiliation Type is required", Toast.LENGTH_LONG).show();
-      return;
+      return null;
     } else if (selectedItem.toString().equals("Military")) {
```

Comment:
@sdownie yes, to support this feature we should provide an interface to the user, so they should implement it with their custom scope. This interface should have only one method which is `String getScopeName()`. By doing this, the api is strongly typed and we support custom scopes. makes sense?